server_test.go

```diff
@@ -16,69 +16,58 @@ import (
 )
 
 func TestStubServer(t *testing.T) {
-	server := getStubServer(t)
-
-	req := httptest.NewRequest("POST", "https://stripe.com/v1/charges",
-		bytes.NewBufferString("amount=123&currency=usd"))
-	req.Header.Set("Authorization", "Bearer sk_test_123")
-	w := httptest.NewRecorder()
-	server.HandleRequest(w, req)
-
-	resp := w.Result()
-	body, err := ioutil.ReadAll(resp.Body)
-	assert.NoError(t, err)
+	resp, body := sendRequest(t, "POST", "/v1/charges",
+		"amount=123&currency=usd", getDefaultHeaders())
 
 	assert.Equal(t, http.StatusOK, resp.StatusCode)
 
 	var data map[string]interface{}
-	err = json.Unmarshal(body, &data)
+	err := json.Unmarshal(body, &data)
 	assert.NoError(t, err)
 	_, ok := data["id"]
 	assert.True(t, ok)
-}
 
-func TestStubServer_SetsSpecialHeaders(t *testing.T) {
-	server := getStubServer(t)
+	// Ensure error response has the correct shape
```

Comment:
Maybe this should be in a separate test?

Reply:
https://github.com/stripe/stripe-mock/pull/32/commits/e5dfd59b237c2cbdf520d0e3c3b784739cad5969